### PR TITLE
Fix signature verify

### PR DESCRIPTION
**Pull request by Akagi201**
This is a temporary fix, further discussion on https://github.com/Commit-Boost/commit-boost-client/issues/157

## Summary by CodeRabbit

- **New Features**
	- Enhanced visibility of fields in the `PreconferElection` struct for easier access.
	- Added new methods in the `ProposerApi` for managing constraints and elections: `set_constraints` and `elect_preconfer`.
	- Added new development dependencies for improved functionality.

- **Bug Fixes**
	- Simplified logic in the `verify_signed_commit_boost_message` function for better performance.

- **Tests**
	- Introduced a new test module to verify the signature of `PreconferElection` messages.

### CODE REVIEW

File: crates/common/src/api/constraints_api.rs [hunk 1]
```diff
@@ -44,13 +44,13 @@ impl SignedPreconferElection {
...
     /// Slot this delegation is valid for.
-    slot_number: u64,
+    pub slot_number: u64,
     /// Chain ID of the chain this election is for.
-    chain_id: u64,
-    // The gas limit specified by the proposer that the preconfer must adhere to.
-    gas_limit: u64,
+    pub chain_id: u64,
+    /// The gas limit specified by the proposer that the preconfer must adhere to.
+    pub gas_limit: u64,
```
[the remark this answers is not in the document]
Comment (Akagi201) [reply crates/common/src/api/constraints_api.rs:53]: ignore

File: crates/utils/src/signing.rs [hunk 2]
```diff
@@ -46,12 +46,13 @@ pub fn verify_signed_commit_boost_message<T: HashTreeRoot>(
     message: &mut T,
     signature: &BlsSignature,
     public_key: &BlsPublicKey,
-    slot_hint: Option<Slot>,
+    _slot_hint: Option<Slot>,
```
Comment (zsluedem) [inline crates/utils/src/signing.rs]: remove it if it is not neccesary.

File: crates/utils/src/signing.rs [hunk 3]
```diff
@@ -46,12 +46,13 @@ pub fn verify_signed_commit_boost_message<T: HashTreeRoot>(
...
     root_hint: Option<Root>,
     context: &Context,
 ) -> Result<(), Error> {
-    let fork_version = slot_hint.map(|slot| context.fork_version_for(context.fork_for(slot)));
```
Comment (zsluedem) [inline crates/utils/src/signing.rs:53]: Good catch. But I think this is the implemetation invented by ourselves. Helix original algorithm doesn't use slot.

Comment (zsluedem) [reply crates/utils/src/signing.rs:53]: You can see https://github.com/lu-bann/helix/pull/15/files#diff-3b2d9daf9eed98dd27c340ab94524e841d07f6dbf9fc729e15875b448c412d71R34-R43

File: crates/utils/src/signing.rs (hunk 2 above)
Comment (Akagi201) [reply crates/utils/src/signing.rs]: it's a temporary fix, other helix verification method use slot.

Comment (zsluedem) [reply crates/utils/src/signing.rs]: No. The original algorithm doesn't use slot.

Comment (Akagi201) [reply crates/utils/src/signing.rs]: you can search fork version uses in the source code, there are two ways to fill the fork version

Comment (zsluedem) [reply crates/utils/src/signing.rs]: The way you see using slot is for block signing

**Pull request merged**